cmd/snap/cmd_run.go

```diff
@@ -216,13 +217,20 @@ func snapRunHook(snapName, snapRevision, hookName string) error {
 }
 
 func runSnapConfine(info *snap.Info, securityTag, snapApp, command, hook string, args []string) error {
+	snapConfine := filepath.Join(dirs.DistroLibExecDir, "snap-confine")
+	if !osutil.FileExists(snapConfine) {
+		if hook != "" {
+			logger.Noticef("WARNING: skipping runnning snap/hook %s/%s: missing snap-confine", info.Name(), hook)
+			return nil
+		}
+		return fmt.Errorf(i18n.G("missing snap-confine: is your snapd too old?"))
```

Comment:
Perhaps we can be more prescriptive:

"missing snap-confine: try updating the snapd package"